src/lib/arch/riscv/Instruction.cc

```diff
@@ -165,12 +157,32 @@ const std::vector<uint16_t>& Instruction::getSupportedPorts() {
   return supportedPorts_;
 }
 
+void Instruction::setExecutionInfo(const ExecutionInfo& info) {
+  if (isInstruction(InsnIdentifier::isLoadMask) ||
+      isInstruction(InsnIdentifier::isStoreMask)) {
+    lsqExecutionLatency_ = info.latency;
+  } else {
+    latency_ = info.latency;
+  }
+  stallCycles_ = info.stallCycles;
+  supportedPorts_ = info.ports;
+}
+
 const InstructionMetadata& Instruction::getMetadata() const { return metadata; }
 
 const Architecture& Instruction::getArchitecture() const {
   return architecture_;
 }
 
+InstructionException Instruction::getException() const { return exception_; }
+
+bool Instruction::isAtomic() const {
```

Comment:
If we move back into public, we should bump up `isAtomic` and `isFloat` in file to be with the other instruction identifier getters